src/main/java/com/example/onlinebookstore/service/shopping/cart/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package com.example.onlinebookstore.service.shopping.cart;
+
+import com.example.onlinebookstore.dto.cart.items.CreateCartItemRequestDto;
+import com.example.onlinebookstore.dto.shopping.cart.CartItemUpdateRequestDto;
+import com.example.onlinebookstore.dto.shopping.cart.ShoppingCartDto;
+import com.example.onlinebookstore.exception.EntityNotFoundException;
+import com.example.onlinebookstore.mapper.ShoppingCartMapper;
+import com.example.onlinebookstore.model.Book;
+import com.example.onlinebookstore.model.CartItem;
+import com.example.onlinebookstore.model.ShoppingCart;
+import com.example.onlinebookstore.model.User;
+import com.example.onlinebookstore.repository.BookRepository;
+import com.example.onlinebookstore.repository.CartItemRepository;
+import com.example.onlinebookstore.repository.ShoppingCartRepository;
+import com.example.onlinebookstore.repository.UserRepository;
+import jakarta.transaction.Transactional;
+import java.util.HashSet;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final UserRepository userRepository;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+
+    @Override
+    @Transactional
```

Comment:
I don't sure, but I think you will have more requests to DB with Transactional. Instead this, you can use EntityGraph in repository method. This approach definitely generate one SQL query.